app/helpers/products_helper.rb

```diff
@@ -1,4 +1,35 @@
 module ProductsHelper
+  # returns zero for all values if test is false
+  def checklist_status_values(test)
+    passing = failing = total = 0
+    return [passing, failing, total] unless test
+    passing = test.num_measures_complete
+    total = test.num_measures
+    failing = total - passing
+    [passing, failing, total]
+  end
+
+  def measure_test_status_values(tests, task_type)
+    passing = failing = not_started = total = 0
+    tasks = []
+    tests.each { |test| tasks << test.tasks.where(_type: task_type) }
+    unless tasks.empty?
+      passing = tasks.count { |task| task.first.status == 'passing' }
+      failing = tasks.count { |task| task.first.status == 'failing' }
+      not_started = tasks.count { |task| task.first.status == 'incomplete' }
+      total = tasks.count
+    end
+    [passing, failing, not_started, total]
+  end
+
+  def filtering_test_status_values(tests)
+    passing = failing = not_started = total = 0
+
+    # add content here when C4 tasks are finalized
+
+    [passing, failing, not_started, total]
```

Comment:
There are a couple of places where we use this construct. Might it be worth making this a small class, rather than an array? @ssayer/ @rdingwell / others; opinions?